source/common/router/router.cc

```diff
@@ -704,8 +704,33 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
   // Hang onto the modify_headers function for later use in handling upstream responses.
   modify_headers_ = modify_headers;
 
+  bool can_use_early_data{false};
+  if (Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.conn_pool_new_stream_with_early_data_and_alt_svc")) {
+    absl::string_view method = downstream_headers_->getMethodValue();
+    // According to safe methods defined in https://www.rfc-editor.org/rfc/rfc7231#section-4.2.1
+    const bool is_safe_method = (method == Http::Headers::get().MethodValues.Get ||
+                                 method == Http::Headers::get().MethodValues.Head ||
+                                 method == Http::Headers::get().MethodValues.Options ||
+                                 method == Http::Headers::get().MethodValues.Trace);
+
+    if (retry_state_ == nullptr) {
+      ENVOY_LOG(warn, "No retry policy is configured. There won't be any fallback for alternate "
```

Comment:
I'm not sure what the Envoy approach to warning is. I wonder if this might be too frequent since it's potentially once for every request?